packages/@sanity/base/src/datastores/grants/documentPairPermissions.ts

```diff
@@ -32,7 +32,17 @@ function getPairPermissions({
   draft,
   published,
   liveEdit,
-}: PairPermissionsOptions): Array<[string, Observable<PermissionCheckResult>] | null> {
+}: PairPermissionsOptions): Array<[string, Observable<PermissionCheckResult>]> {
+  // this was introduced because we ran into a bug where a user with publish
+  // access was marked as not allowed to duplicate a document unless it had a
+  // draft variant. this would happen in non-live edit cases where the document
+  // pair only had a published variant with the draft variant being null.
+  //
+  // note: this should _not_ be used if the draft and published versions should
+  // be considered separately/explicitly in the permissions.
```

Comment:
nit, but wondering if it would be better to move the variables here into the two respective blocks that uses them? It adds a bit of duplication, but think it's justified by increased cohesion and also keeping them out of scope for the other checks, and this comment can be removed.

Reply:
i like it!